scripts/provision_credentials.py

```diff
@@ -0,0 +1,169 @@
+            if e.response['Error']['Code'] == 'ResourceExistsException':
+                logger.info('AWS secret %s already exists.', name)
+            else:
+                raise
+        else:
+            logger.info('AWS secret %s created.', name)
+
+    def _secret_is_stored(self, name):
+        try:
+            response = self.secrets_manager.get_secret_value(SecretId=name)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                return False
+            else:
+                raise
+        try:
+            return response['SecretString'] != ''
+        except KeyError:
+            return False
+
+    def _create_service_account_creds(self, service_account_email):
+        service = get_google_service()
+        key = service.projects().serviceAccounts().keys().create(
+            name='projects/-/serviceAccounts/' + service_account_email, body={}
+        ).execute()
+        logger.info("Successfully created service account key for user '%s'", service_account_email)
+        return parse_google_key(key)
+
+    def _destroy_aws_secrets_manager_secret(self, secret_name):
+        try:
+            response = self.secrets_manager.delete_secret(
+                SecretId=secret_name,
+                ForceDeleteWithoutRecovery=True
+            )
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info('AWS secret %s doesn\'t exist. No changes will be made.', secret_name)
+            else:
+                raise
+        else:
+            assert response['Name'] == secret_name
+            logger.info('Successfully deleted AWS secret %s.', secret_name)
+
+    def _destroy_service_account_creds(self, service_account_email):
+        try:
+            creds = self.secrets_manager.get_secret_value(
+                SecretId=config.secrets_manager_secret_name('indexer', 'google_service_account')
+            )
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info('Secret already deleted, cannot get key_id for %s', service_account_email)
+                return
+            else:
+                raise
+        else:
+            key_id = json.loads(creds['SecretString'])['private_key_id']
+            service = get_google_service()
+            try:
+                service.projects().serviceAccounts().keys().delete(
+                    name='projects/-/serviceAccounts/' + service_account_email + '/keys/' + key_id).execute()
+            except HttpError as e:
+                if e.resp.reason != 'Not Found':
+                    raise
+            logger.info("Successfully deleted service account key with id '%s' for user '%s'",
+                        key_id, service_account_email)
+
+
+if __name__ == "__main__":
+    credentials_provisioner = CredentialsProvisioner()
```

Comment:
This means that AWS credential errors would be reported before command line usage errors. Generally, the instantiation of the helper class could also have other side-effects that require it to occur after the CLI argumemt checking. Most importantly, the instance may need to parameterized with CLI arguments in which case it can only be instantiated after the arguments have been parsed. See below.

Reply:
Good point, I've refactored with your suggestions.